ddtrace/internal/telemetry/writer.py

```diff
@@ -675,56 +576,68 @@ def periodic(self, force_flush=False, shutting_down=False):
             # app-started should be the first event in the batch
-            events = [app_started] + events
+            events = [self._get_event(app_started_payload, TELEMETRY_EVENT_TYPE.STARTED)] + events
 
-        if app_product_change := self._app_product_change():
-            events.append(app_product_change)
+        if products := self._report_products():
+            events.append(self._get_event({"products": products}, TELEMETRY_EVENT_TYPE.PRODUCT_CHANGE))
 
-        if integrations := self._flush_integrations_queue():
-            events.append(self._app_integrations_changed_event(integrations))
+        if ints := self._report_integrations():
+            events.append(self._get_event({"integrations": ints}, TELEMETRY_EVENT_TYPE.INTEGRATIONS_CHANGE))
 
-        if endpoints_payload := self._flush_app_endpoints():
-            events.append(endpoints_payload)
+        if endpoints := self._report_endpoints():
+            events.append(self._get_event(endpoints, TELEMETRY_EVENT_TYPE.ENDPOINTS))
 
-        if configurations := self._flush_configuration_queue():
-            events.append(self._app_client_configuration_changed_event(configurations))
+        if configs := self._report_configurations():
+            events.append(self._get_event({"configuration": configs}, TELEMETRY_EVENT_TYPE.CLIENT_CONFIGURATION_CHANGE))
 
-        if app_dependencies_loaded := self._app_dependencies_loaded_event():
-            events.append(app_dependencies_loaded)
+        if deps := self._report_dependencies():
+            events.append(self._get_event({"dependencies": deps}, TELEMETRY_EVENT_TYPE.DEPENDENCIES_LOADED))
 
-        if shutting_down and (app_closing := self._app_closing_event()):
-            events.append(app_closing)
+        if shutting_down and not self._forked:
+            events.append(self._get_event({}, TELEMETRY_EVENT_TYPE.SHUTDOWN))
 
         # Always include a heartbeat to keep RC connections alive
-        events.append(self._app_heartbeat_event())
+        # Extended heartbeat should be queued after app-dependencies-loaded event. This
+        # ensures that that imported dependencies are accurately reported.
+        if heartbeat_payload := self._report_heartbeat():
+            # Extended heartbeat report dependencies while regular heartbeats report empty payloads
+            events.append(self._get_event(heartbeat_payload, TELEMETRY_EVENT_TYPE.EXTENDED_HEARTBEAT))
+        else:
+            events.append(self._get_event({}, TELEMETRY_EVENT_TYPE.HEARTBEAT))
 
-        # Get any queued events and combine with current batch
-        if queued_events := self._flush_events_queue():
+        # Get any queued events (ie metrics and logs from previous periodic calls) and combine with current batch
+        if queued_events := self._report_events():
             events.extend(queued_events)
 
-        log.debug("Encoding instrumentation telemetry events: %s", ", ".join([e["request_type"] for e in events]))
```

Comment:
Since this log is being deleted, is this information we may need in the future to debug?

Reply:
This log was consolidated with the log on line 101. So instead of two debugs per instrumentation telemetry payloads we will only send one log with all the info